src/oop/static_modifier/hard/q3/MainTest.java

```diff
@@ -0,0 +1,156 @@
+        String wrongTypeMessage = "The %s attribute in the %s class has the wrong type.";
+        String wrongModifierMessage = "The %s attribute in the %s class has the wrong visibility modifier.";
+        assertTrue(testClass.hasField(attributeName), String.format(missingAttributeMessage, className, attributeName));
+        assertTrue(testClass.hasField(attributeName, String.class), String.format(wrongTypeMessage, attributeName, className));
+        assertTrue(testClass.hasModifier(attributeName, "private"), String.format(wrongModifierMessage, attributeName, className));
+        assertTrue(testClass.hasField(staticAttributeName), String.format(missingAttributeMessage, className, staticAttributeName));
+        assertTrue(testClass.hasField(staticAttributeName, int.class), String.format(wrongTypeMessage, staticAttributeName, className));
+        assertTrue(testClass.hasModifier(staticAttributeName, "private"), String.format(wrongModifierMessage, staticAttributeName, className));
+        assertTrue(testClass.hasModifier(staticAttributeName, "static"), String.format(wrongModifierMessage, staticAttributeName, className));
+    }
+
+    @Test
+    @Order(3)
+    public void candyStoreHasRequiredConstructor() {
+        String missingConstructorMessage = "The %s class is missing a required constructor. Make sure that it is named correctly and has the correct parameters.";
+        String wrongAccessModifier = "The %s class constructor has the wrong visibility modifier. Make sure that it is visible from all other classes.";
+        Class<?>[] constructorArgs = new Class[]{String.class};
+        assertTrue(testClass.hasConstructor(constructorArgs), String.format(missingConstructorMessage, className));
+        assertTrue(testClass.hasModifier(constructorArgs, "public"), String.format(wrongAccessModifier, className));
+    }
+
+    @Test
+    @Order(1)
+    @Tag("dependency1")
+    public void gummiesIsInitializedCorrectly() {
+        _assertEquals(initialGummies, testClass.getFieldValue(null, staticAttributeName),
+                "Your " + staticAttributeName + " static attribute is not initialized correctly.");
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of("CandyLand", initialGummies - 30),
+                Arguments.of("GummyGrove", initialGummies - 50),
+                Arguments.of("SugarShack", initialGummies / 2),
+                Arguments.of("SweetSpot", initialGummies - 75),
+                Arguments.of("TreatTown", initialGummies / 3),
+                Arguments.of("BonbonBoulevard", initialGummies - 60),
+                Arguments.of("Lollipop Lane", initialGummies / 4),
+                Arguments.of("Jellybean Junction", initialGummies - 15),
+                Arguments.of("Confectionery Corner", initialGummies / 5),
+                Arguments.of("Dulce Drive", initialGummies - 90)
```

Comment:
add test cases where the number of awarded gummies is too high so the subtraction doesn't trigger